src/types/plugin-input.ts

```diff
@@ -0,0 +1,28 @@
+import { SupportedEvents, SupportedEventsU } from "./context";
+import { StaticDecode, Type as T } from "@sinclair/typebox";
+
+export interface PluginInputs<T extends SupportedEventsU = SupportedEventsU, TU extends SupportedEvents[T] = SupportedEvents[T]> {
+  stateId: string;
+  eventName: T;
+  eventPayload: TU["payload"];
+  settings: StartStopSettings;
+  authToken: string;
+  ref: string;
+}
+
+export const startStopSchema = T.Object({
```

Comment:
Should have defaults

Reply:
Aren't the defaults passed in via the kernel, I thought they were in `generateConfiguration()`, no? I will update

Reply:
Impossible for kernel to generate defaults for every possible plugin in the future.

Reply:
that makes sense